infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/provision/ProxySettingsProvisioner.java

```diff
@@ -60,13 +61,15 @@ public void provision(KubernetesEnvironment k8sEnv, RuntimeIdentity identity)
     TracingTags.WORKSPACE_ID.set(identity::getWorkspaceId);
 
     if (!proxyEnvVars.isEmpty()) {
-      for (PodData pod : k8sEnv.getPodsData().values()) {
-        pod.getSpec()
-            .getContainers()
-            .forEach(
-                container ->
-                    proxyEnvVars.forEach((k, v) -> container.getEnv().add(new EnvVar(k, v, null))));
-      }
+      k8sEnv
+          .getPodsData()
+          .entrySet()
+          .stream()
+          .filter(entry -> !entry.getKey().equals(JWT_PROXY_POD_NAME))
```

Comment:
1. Please consider adding one like comment why here we ignore JWT_PROXY pod.

2. Please consider adding test case for this.

Reply:
Why i am always modifying classes which doesn;t have tests at all :rofl: ?

Reply:
Cause you're a lucky man! :rofl: